be/src/storage/persistent_index.h

```diff
@@ -269,6 +275,15 @@ class PersistentIndex {
     // |old_values|: return old values if key exist, or set to NullValue if not
     Status erase(size_t n, const void* keys, IndexValue* old_values);
 
+    // batch replace
+    // |n|: size of key/value array
+    // |keys|: key array as raw buffer
+    // |values|: value array
+    // |src_rssid|: rssid array
+    // |failed|: return not match rowid
+    Status try_replace(size_t n, const void* keys, const IndexValue* values, const std::vector<uint32_t>& src_rssid,
+                       std::vector<uint32_t>* failed);
```

Comment:
Add ut for this

Reply:
> Add ut for this

done, change in https://github.com/StarRocks/starrocks/pull/4174/commits/29d627d709a5a85f7ec601013de5412fe544f2e9